docs/user_guides/fs/transformation_functions.md

```diff
@@ -0,0 +1,163 @@
+
+# Transformation Functions    
+
+In AI systems, [transformation functions](https://www.hopsworks.ai/dictionary/transformation) modify data used for machine learning applications, often to improve performance. The [taxonomy of data transformations](../../concepts/mlops/data_transformations.md) introduces three types of data transformation prevalent in all AI systems. Hopsworks offers simple Python APIs to define custom transformation functions. These can be used along with [feature groups](./feature_group/index.md) and [feature views](./feature_view/overview.md) to create [on-demand transformations](./feature_group/on_demand_transformations.md) and [model-dependent transformations](./feature_view/model-dependent-transformations.md), producing modular AI pipelines that are skew-free.
+
+## Creation
+
+User-defined transformation functions can be created in Hopsworks using the [`@udf`](http://docs.hopsworks.ai/hopsworks-api/{{{hopsworks_version}}}/generated/api/udf/) decorator. These functions should be designed as Pandas functions, meaning they must take input features as a [Pandas Series](https://pandas.pydata.org/docs/reference/api/pandas.Series.html) and return either a Pandas Series or a [Pandas DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html). Hopsworks automatically executes the defined transformation function as a [`pandas_udf`](https://spark.apache.org/docs/3.1.2/api/python/reference/api/pyspark.sql.functions.pandas_udf.html) in a PySpark application and as Pandas functions in Python clients.
```

Comment:
What about a Pandas DataFrame as input?

Reply:
Currently I have not added the support of pandas Dataframes as inputs. Each input features is a pandas Series. 

Input of a pandas DataFrame in pandas UDF in Spark is only supported if the input column is a complex type(`Struct` Type).

I will add the support. I though that we did not support Struct Types, but form the documentation I saw that we support `Structs `in both offline and online tables.